src/DeviceDiscoveryUI/DDDevicePickerViewController.cs

```diff
@@ -0,0 +1,54 @@
+//
+// Custom methods for DDDevicePickerViewController
+//
+// Authors:
+//	Israel Soto  <issoto@microsoft.com>
+//
+// Copyright 2022 Microsoft Corporation.
+//
+
+#nullable enable
+
+using ObjCRuntime;
+using Foundation;
+using Network;
+using System;
+
+using OS_nw_endpoint = System.IntPtr;
+using OS_nw_error = System.IntPtr;
+
+namespace DeviceDiscoveryUI {
+
+	public partial class DDDevicePickerViewController {
+		public delegate void DevicePickerCompletionHandler (NWEndpoint? endpoint, NWError? error);
+
+		public static bool IsSupported (NWBrowserDescriptor browseDescriptor, NWParameters? parameters)
+		{
+			if (browseDescriptor == null)
+				throw new ArgumentNullException (nameof (browseDescriptor));
+
+			return _IsSupported (browseDescriptor.Handle, parameters?.Handle ?? IntPtr.Zero);
```

Comment:
This entire method could in fact be (this would not need the null check above either):

```suggestion
			return _IsSupported (browseDescriptor.GetNonNullHandle (nameof (browseDescriptor)), parameters.GetHandle ());
```

See:

https://github.com/xamarin/xamarin-macios/blob/3b1037e56f1cc1943fc4ea5af15ac17efcd24466/src/ObjCRuntime/INativeObject.cs#L23-L37